### DISCUSSION TITLE
New Name for opentsdb package

### DISCUSSION THREAD

**Discussion by MikeNeilson** [opendcs/opendcs#89] [Polls]
[OpenTSDB](http://opentsdb.net/) is the name of a very popular NoSQL time series database. By using that name for our Timeseries database we my cause confusion, and may, more likely does, run afoul of trademark concerns. Instead of waiting for the OpenTSDB organization so send us some sort of cease-and-desist and force us to rush it would be better to just pick a name now and alter that which needs altering.

If we ever support using OpenTSDB it would make sense to have packages with that name as that would not cause confusion, but ours is an entirely different database.

I'm junk with names, feel free to add/suggest options.

I refer to it as the "reference" database as it has no dependencies on and external system and will let us do more automated testing against a live database.

@opendcs/opendcs-core-devs @opendcs/opendcs-dcpmon-devs @opendcs/opendcs-triange

**Comment by mikecovesw** [comment]
CWMS Lite
Open Hydro DB

The trouble with calling it "simple" is that it may evolve to be quite complex.
WaterTSDB is fine.

**Reply by DanielTOsborne** [reply]
It might be too specific of a name, but I like Open Hydro DB.

**Comment by agilmore2** [comment]
DCSTSDB

I like CWMSLite, since I hear it called that a lot.

Too bad OpenTSDB uses unix epoch seconds, I'd be more interested if they
were signed 64 bits. ;)

**Reply by MikeNeilson** [reply]
It should really just be a proper timestamp column type.

**Comment by mikecovesw** [comment]
I'm not sure I understand your comment. There are perhaps a few places 
left in OpenDCS (including our OpenTSDB) that use a java 32-bit signed 
int unix epoch seconds, but I had been trying to expand those to 64-bit 
Java msecs. But what does that have to do with the name?

- Mike M

On 8/22/22 10:21 AM, agilmore2 wrote:
> DCSTSDB
>
> I like CWMSLite, since I hear it called that a lot.
>
> Too bad OpenTSDB uses unix epoch seconds, I'd be more interested if they
> were signed 64 bits. ;)
>
> On Mon, Aug 22, 2022, 08:07 Mike Maloney ***@***.***> wrote:
>
> > CWMS Lite
> > Open Hydro DB
> >
> > The trouble with calling it "simple" is that it may evolve to be quite
> > complex.
> > WaterTSDB is fine.
> >
> > —
> > Reply to this email directly, view it on GitHub
> > 
> <https://github.com/opendcs/opendcs/discussions/89#discussioncomment-3448028>,
> > or unsubscribe
> > 
> <https://github.com/notifications/unsubscribe-auth/ABUI5STIRMFH4IKX4YYEQRTV2OCSPANCNFSM57B5KQFQ>
> > .
> > You are receiving this because you are on a team that was 
> mentioned.Message
> > ID: ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/opendcs/opendcs/discussions/89#discussioncomment-3448130>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABTDKHXVEKWBCYMPZXZBOALV2OEFJANCNFSM57B5KQFQ>.
> You are receiving this because you are on a team that was 
> mentioned.Message ID: 
> ***@***.***>
>
-- 
Michael Maloney
Cove Software, LLC
Tel: (410) 241-5248
***@***.***
Web:www.covesw.com

**Reply by MikeNeilson** [reply]
> I'm not sure I understand your comment. There are perhaps a few places left in OpenDCS (including our OpenTSDB) that use a java 32-bit signed int unix epoch seconds, but I had been trying to expand those to 64-bit Java msecs. But what does that have to do with the name? - Mike M
> On 8/22/22 10:21 AM, agilmore2 wrote: DCSTSDB I like CWMSLite, since I hear it called that a lot. Too bad OpenTSDB uses unix epoch seconds, I'd be more interested if they were signed 64 bits. ;) On Mon, Aug 22, 2022, 08:07 Mike Maloney ***@***.***> wrote: > CWMS Lite > Open Hydro DB > > The trouble with calling it "simple" is that it may evolve to be quite > complex. > WaterTSDB is fine. > > 

Well, it's proof of the confusion that can happen by using the name "opentsdb".

I believe @agilmore2 was referring to the other OpenTSDB, time in which is limited to the Unix Epoch and 13 actual digits.
See: http://opentsdb.net/docs/build/html/user_guide/writing/index.html#data-specification

**Comment by rma-psmorris** [comment]
My $0.02. While being time series centric, the schema has more than just time series. I suggest "OpenDCSDB". Identifies what it is for, "OpenDCS", and the fact that it is a database, "DB".

**Comment by agilmore2** [comment]
I meant the nosql one.

**Comment by agilmore2** [comment]
Open Hydro DB sounds HDB compatible, rather than CMWS compatible. ;)

**Reply by MikeNeilson** [reply]
true, though I think it should be both compatible. It seems HDB requires more metadata exposed in the GUI to determine timeseries, we're going to need that soon for CWMS with our versioned time series and other things (like forecast data.)

We also talked about merging the CWMS and HDB database. The OpenDCS database seems like a good candidate to start identifying those concepts so that:
1. We get put them in and development is easier for new people (much easier to stand up a PostgresSQL db)
2. We can all look at those abstractions together can share thoughts to make them better.

Whether it becomes *the* shared db design is immaterial, I think push that goal helps us either way.

**Comment by MikeNeilson** [comment]
I've reset the poll options with the names presented in the comments. We are at the limit of poll options though. But I think that captures everything.

**Comment by MikeNeilson** [comment]
Also I should note, once we actually settle on a name, there will definitely be some staging time before actual changes. Since we keep fully qualified class names in configuration it should definitely be considered a breaking change and likely require a bump to 8.0.0 when we make it happen.

**Comment by MikeNeilson** [comment]
This poll has been up for quite a while and the majority of votes are for OpenHydroDB.
I'm going to say that's it and we can then start down that path.
